spec/lib/openid_token_proxy/client_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'spec_helper'
+
+require 'securerandom'
+
+RSpec.describe OpenIDTokenProxy::Client do
+  subject { described_class.new config }
+  let(:config) { OpenIDTokenProxy::Config.new }
+
+  describe '#config' do
+    it 'defaults to Config.instance' do
+      client = described_class.new
+      expect(client.config).to eq OpenIDTokenProxy::Config.instance
+    end
+
+    it 'may be given explicitly' do
+      expect(subject.config).to eq config
+    end
+  end
+
+  describe '#authorization_uri' do
+    it 'may be explicitly set through environment' do
+      stub_env('OPENID_AUTHORIZATION_URI', 'from env')
+      expect(subject.authorization_uri).to eq 'from env'
+    end
+
+    context 'when resource given' do
+      it 'defaults to OpenID authorization URI' do
+        authorization_uri = subject.authorization_uri
+        expect(authorization_uri).not_to be_blank
+        expect(authorization_uri).not_to include 'resource='
+        expect(authorization_uri).not_to include 'redirect_uri='
```

Comment:
This might just be my lack of understanding on how OpenID works, but it's not clear to me how these three expectations ensure that the authorization URI is an OpenID one. Not sure if there's a better way, though.